instrumentation/runtime-metrics/runtime-metrics-java17/library/src/main/java/io/opentelemetry/instrumentation/runtimemetrics/java17/RuntimeMetrics.java

```diff
@@ -55,11 +58,11 @@ public final class RuntimeMetrics implements Closeable {
 
       // Set up metrics gathered by JMX
       if (!disableJmx) {
-        BufferPools.registerObservers(openTelemetry);
-        Classes.registerObservers(openTelemetry);
-        Cpu.registerObservers(openTelemetry);
-        MemoryPools.registerObservers(openTelemetry);
-        Threads.registerObservers(openTelemetry);
+        observables.addAll(BufferPools.registerObservers(openTelemetry));
+        observables.addAll(Classes.registerObservers(openTelemetry));
+        observables.addAll(Cpu.registerObservers(openTelemetry));
+        observables.addAll(MemoryPools.registerObservers(openTelemetry));
+        observables.addAll(Threads.registerObservers(openTelemetry));
         GarbageCollector.registerObservers(openTelemetry);
```

Comment:
Nit for next PR (so that we can merge this one finally 😅 ): the GC metrics class could also return a closeable that'll unregister the `NotificationListener`

Reply:
ok that's a good idea. I'll make an issue for that so we don't lose track of it.

Reply:
Created an issue here: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8475